src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -116,20 +114,37 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
         ChatOptions? options = null,
         bool? useNativeJsonSchema = null,
         CancellationToken cancellationToken = default)
-        where T : class
     {
         _ = Throw.IfNull(chatClient);
         _ = Throw.IfNull(chatMessages);
         _ = Throw.IfNull(serializerOptions);
 
         serializerOptions.MakeReadOnly();
 
-        var schemaNode = AIJsonUtilities.CreateJsonSchema(
+        var schemaElement = AIJsonUtilities.CreateJsonSchema(
             type: typeof(T),
             serializerOptions: serializerOptions,
             inferenceOptions: _inferenceOptions);
 
-        var schema = JsonSerializer.Serialize(schemaNode, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
+        var isWrappedInObject = false;
+        var schema = JsonSerializer.Serialize(schemaElement, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
+        if (!SchemaRepresentsObject(schemaElement))
+        {
+            // For non-object-representing schemas, we wrap them in an object schema, because all
+            // the real LLM providers today require an object schema as the root. This is currently
+            // true even for providers that support native structured output.
+            isWrappedInObject = true;
+            schema = $$"""
+                {
+                    "$schema": "https://json-schema.org/draft/2020-12/schema",
+                    "type": "object",
+                    "properties": {
+                        "data": {{schema}}
```

Comment:
`schema.ToString()` doesn't product valid JSON, unfortunately.

Reply:
There's a `JsonElement.GetRawText()` method but that could be using a radically different indentation settings compared to the envelope. I would probably just wrap in a `JsonObject` and serialize using indentation settings present in the JSO.

Reply:
You mean going back to the original version that deserializes `schemaElement` into a `JsonObject` using `JsonSerializer.Deserialize`?

Reply:
Yes, rather than deserializing you can wrap the element with a `JsonValue.Create` call.

Reply:
Can you be really specific? It sounds like you mean:

```cs
JsonValue.Create(schemaElement)
```

... where `schemaElement` is the value returned by `AIJsonUtilities.CreateJsonSchema`. But this fails at runtime (`ThrowInvalidOperationException_NodeElementCannotBeObjectOrArray`). Do you mean something different?

Reply:
Do you mean `JsonObject.Create(schemaElement)`?

Reply:
Ugh. It's a known issue it seems: https://github.com/dotnet/runtime/issues/70427#issuecomment-1150960366 but there is a relatively simple workaround:
```c#
static JsonNode? FromJsonElement(JsonElement element)
{
    return element.ValueKind switch
    {
        JsonValueKind.Array => JsonArray.Create(element),
        JsonValueKind.Object => JsonObject.Create(element),
        _ => JsonValue.Create(element)
    };
}
```

Reply:
Thanks. Done like this.